### DISCUSSION TITLE
Do u need a rust target?

### DISCUSSION THREAD

**Discussion by xuesu** [antlr/antlr4#3416] [Ideas]
I wonder if you have/need to write a rust target

**Comment by ericvergnaud** [comment]
Wouldn't that be for Rust users to say ?
ANTLR4 is OSS, now has 250+ contributors, and welcomes anyone willing to build a Rust target.

**Comment by KvanTTT** [comment]
Sure, why not? FYI, we have a related issue: https://github.com/antlr/antlr4/issues/1839 Take attention to it if you are going to implement Rust target.

**Comment by parrt** [comment]
Oh man, I hesitate to add yet another target as I just finished a deep dive into the testing rig across targets. ugh.

**Comment by Code7R** [comment]
But somebody started creating one. Looks like having being abandoned in the last months, though. Any chance of getting this added?

https://github.com/rrevenantt/antlr4
https://github.com/rrevenantt/antlr4rust

**Reply by jimidle** [reply]
I think the issue is as @parrt says to be honest. It isn't just whether anyone has the time to write it, but it adds a lot more maintenance and release and test and so on. 

I would be willing to do it if someone wanted to pay me and Terence was willing to accept a new target. The go target should not need much maintenance once I have finished optimizing memory and some final micro-optimizations, so there should be time.

A new target is a lot of work. I think people start out on it thinking they will just copy the Java runtime, but that usually gets abandoned once people realize that it isn't quite as simple as that. Memory management can be an issue for instance.

Perhaps you can use one of the other targets and communicate/call in to recognizer from Rust? We don't have to use the same language for everything ;)

**Reply by skull-squadron** [reply]
That try may make a good starting point though.

Exactly. It's never about throwing an implementation over "the wall"(tm) once. It's about on-going subject manager expertise support. There's plenty of competency within the Rust community, so that shouldn't be the show stopper. The irreducible issue is who can volunteer the time and responsiveness to take point on this. (Aren't there like a million Rust Meta PEs, Alphabet SREs, and startup SWEs now? ;)

## 🦀
